x-pack/plugins/enterprise_search/common/types/connectors.ts

```diff
@@ -27,6 +27,8 @@ export interface CustomScheduling {
 
 export type ConnectorCustomScheduling = Record<string, CustomScheduling | null>;
 
+export type ConnectorPreferences = Record<string, unknown>;
```

Comment:
Should update this type too :)

Reply:
@sphillipse Would something like this work? (keeping in mind that there may be more fields added in the future)

```
export interface Preferences {
  extract_full_html: boolean;
}
export type ConnectorPreferences = Preferences | Record<string, unknown>;
```

Reply:
Didn't see your edit, will add. Cheers!

Reply:
If you wanted to do it in an extensible way, you'd do something like:
```
export interface ConnectorPreferences extends Record<string, unknown> {
  extract_full_html: boolean | null;
}
```

Your way is almost correct, but means it's either an object with the defined properties, or an object of shape Record<string, unknown>. In the second case, `extract_full_html` could be anything--so you lose the advantage of typing. Extending it gets rid of that issue.

Reply:
I like this extensible way so I'm going to use it